services/secondary-storage/server/src/main/java/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java

```diff
@@ -1997,6 +1997,17 @@ private void savePostUploadPSK(String psk) {
         }
     }
 
+    protected String getSnapshotFilepathForDelete(String path, String snapshotName) {
+        String finalPath = path + "/*" + snapshotName + "*";
+        if (!path.endsWith(snapshotName)) {
+            return finalPath;
+        }
+        if (s_logger.isDebugEnabled()) {
+            s_logger.debug(String.format("Snapshot file %s is present in the same name directory %s. Deleting the directory", snapshotName, path));
+        }
+        return path;
+    }
```

Comment:
@shwstppr 
would it be safer to check if the path is a directory and there is no files in the directory before deletion ?

Reply:
@weizhouapache directory check be added but we always expect it to be a directory anyway as we append the `/*SNAPSHOT_NAME*

And there will always be files (I checked  VMware, Xen and KVM. Snapshots are placed inside the same name folder **only for VMWare**). Example

```
[root@sl-nestednfs snapshots]# tree 2/10/
2/10/
└── fd57cb44-0559-490c-998d-5313eec52cef
    ├── fd57cb44-0559-490c-998d-5313eec52cef-disk0.vmdk
    ├── fd57cb44-0559-490c-998d-5313eec52cef-disk1.nvram
    └── fd57cb44-0559-490c-998d-5313eec52cef.ovf
```

Reply:
I suppose we need to cover both cases, to check if we're deleting files inside a folder (like in vmware), but also for kvm where snapshots arne't in a folder. XenServer/XCP-ng we need to check.

Reply:
@weizhouapache @shwstppr , we are dealing with a `String`, not a `File`/directory. I think this is ok, in this place. checking for existence should not happen in this method.